File: packages/domain/lib/src/models/skill.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'skill.freezed.dart';
part 'skill.g.dart';

enum SkillCategory { balance, strength, bodyweight }

@freezed
class Skill with _$Skill {
  const factory Skill({
    required String id,
    required String name,
    required String description,
    required SkillCategory category,
  }) = _Skill;

  factory Skill.fromJson(Map<String, dynamic> json) => _$SkillFromJson(json);
}

File: apps/ios_app/Sources/Domain/Models/Skill.swift
import Foundation

enum SkillCategory: String, Codable, CaseIterable {
    case balance
    case strength
    case bodyweight
}

struct Skill: Identifiable, Hashable, Codable, Sendable {
    let id: String
    let name: String
    let description: String
    let category: SkillCategory
    var prescriptionType: PrescriptionType = .duration
}
